queue/priority_queue.go

```diff
@@ -42,50 +39,66 @@ type Item interface {
 
 type priorityItems []Item
 
+func (items *priorityItems) swap(i, j int) {
+	(*items)[i], (*items)[j] = (*items)[j], (*items)[i]
+}
+
+func (items *priorityItems) pop() Item {
```

Comment:
Good point. I used the pointer receiver in new functions merely for consistency (all existing methods in `priorityItems` use this approach). Also, in the case of `push()` and `pop()` the length of the slice gets changed, thus the receiver must still be a pointer as far as I understand (http://blog.golang.org/slices#TOC_5.).